python/ray/tune/tests/test_tune_restore.py

```diff
@@ -314,6 +314,48 @@ def testResourceUpdateInResume(self):
         )
         assert len(analysis.trials) == 27
 
+    # Unfinished trials' resources should be updated.
+    def testConfigUpdateInResume(self):
+        os.environ["TUNE_MAX_PENDING_TRIALS_PG"] = "1"
+
+        config = dict(
+            num_samples=1,
+            fail_fast=True,
+            config={
+                "test": tune.grid_search([1, 2, 3]),
+                "test2": tune.grid_search([1, 2, 3]),
```

Comment:
Nit for the test, can we use different parameter ranges for the different parameters? E.g. 
```suggestion
                "test": tune.grid_search([1, 2, 3]),
                "test2": tune.grid_search([4, 5, 6, 7]),
```

and overwrite with different values as well. Basically to make sure that we we don't conflate parameter overwrites

Reply:
done